BossMod/Components/CastHint.cs

```diff
@@ -36,4 +49,78 @@ public override void OnCastFinished(BossModule module, Actor caster, ActorCastIn
                 _casters.Remove(caster);
         }
     }
+
+    public class CastInterruptHint : CastCounter
+    {
+        public bool Canbeinterrupted;
+        public bool Canbestunned;
+        private List<Actor> _casters = new();
+        public IReadOnlyList<Actor> Casters => _casters;
+        public bool Active => _casters.Count > 0;
+
+        public CastInterruptHint(ActionID action, bool canbeinterrupted = true, bool canbestunned = false) : base(action)
+        {
+            Canbeinterrupted = canbeinterrupted;
+            Canbestunned = canbestunned;
+        }
+
+        public override void AddGlobalHints(BossModule module, GlobalHints hints)
+        {
+            foreach (var caster in _casters)
+            {
+                if (Active && Canbeinterrupted && !Canbestunned)
+                    hints.Add($"Interrupt {caster.Name}!");
+                if (Active && !Canbeinterrupted && Canbestunned)
+                    hints.Add($"Stun {caster.Name}!");
+                if (Active && Canbeinterrupted && Canbestunned)
+                    hints.Add($"Interrupt or stun {caster.Name}!");
+            }
+        }
+
+        public override void OnCastStarted(BossModule module, Actor caster, ActorCastInfo spell)
+        {
+            if (spell.Action == WatchedAction)
+                _casters.Add(caster);
+        }
+
+        public override void OnCastFinished(BossModule module, Actor caster, ActorCastInfo spell)
+        {
+            if (spell.Action == WatchedAction)
+                _casters.Remove(caster);
+        }
+
+        public override void AddAIHints(BossModule module, int slot, Actor actor, PartyRolesConfig.Assignment assignment, AIHints hints)
+        {
+            base.AddAIHints(module, slot, actor, assignment, hints);
+            foreach (var caster in _casters)
+            {
+                if (Active && Canbeinterrupted && actor.Role == Role.Tank && Service.ClientState.LocalPlayer?.Level >= 18)
+                    hints.PlannedActions.Add((ActionID.MakeSpell(WAR.AID.Interject), caster, 1, false));
+                if (Active && Canbeinterrupted && actor.Class.GetClassCategory() == ClassCategory.PhysRanged && Service.ClientState.LocalPlayer?.Level >= 24)
```

Comment:
You can't access ClientState from components, because it won't be available in UIDev. Everything that is needed for components should be part of the world state.

Reply:
Ok, I've added Level field to Actor.

Reply:
In fact, this is wrong - it should not hardcode all possible interrupt actions - rather, it should set appropriate Enemy.ShouldBeInterrupted (and new ShouldBeStunned) flag, and let autorotation modules handle it.

Reply:
I can change it, but then every autorotation needs support for shouldbeinterrupted/stunned I guess

Reply:
Yep, and that's exactly what they should do